### DISCUSSION TITLE
docker compose -f docker-compose-CN.yml up -d not success

### DISCUSSION THREAD

**Discussion by glider2019cn** [infiniflow/ragflow#1167] [General]
Hi All,

At beginning, I have changed RAGFLOW_VERSION=v0.7.0, then run docker compose -f docker-compose-CN.yml up -d , 
it is not working until I run sudo, then start working.
no error, 
 sudo docker compose -f docker-compose-CN.yml up -d
[sudo] password for tac:
WARN[0000] The "MACOS" variable is not set. Defaulting to a blank string.
[+] Running 67/41
 ✔ mysql 11 layers [⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿]      0B/0B      Pulled                                                                                                                                                 11.2s
 ✔ minio 7 layers [⣿⣿⣿⣿⣿⣿⣿]      0B/0B      Pulled                                                                                                                                                      23.0s
 ✔ es01 10 layers [⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿]      0B/0B      Pulled                                                                                                                                                   56.0s
 ✔ ragflow 26 layers [⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿]      0B/0B      Pulled                                                                                                                              1014.9s
 ✔ redis 8 layers [⣿⣿⣿⣿⣿⣿⣿⣿]      0B/0B      Pulled                                                                                                                                                     15.1s

[+] Running 10/10
 ✔ Network docker_ragflow      Created                                                                                                                                                                   0.1s
 ✔ Volume "docker_minio_data"  Created                                                                                                                                                                   0.0s
 ✔ Volume "docker_redis_data"  Created                                                                                                                                                                   0.0s
 ✔ Volume "docker_esdata01"    Created                                                                                                                                                                   0.0s
 ✔ Volume "docker_mysql_data"  Created                                                                                                                                                                   0.0s
 ✔ Container ragflow-mysql     Healthy                                                                                                                                                                   2.2s
 ✔ Container ragflow-redis     Started                                                                                                                                                                   2.2s
 ✔ Container ragflow-minio     Started                                                                                                                                                                   2.2s
 ✔ Container ragflow-es-01     Healthy                                                                                                                                                                   2.2s
 ✔ Container ragflow-server    Started   

but sudo docker logs -f ragflow-server not show success but show :
udo docker logs -f ragflow-server
[sudo] password for tac:
Traceback (most recent call last):
  File "/ragflow/rag/svr/task_executor.py", line 28, in <module>
Traceback (most recent call last):
  File "/ragflow/api/ragflow_server.py", line 26, in <module>
    from api.db.services.file2document_service import File2DocumentService
  File "/ragflow/api/db/services/__init__.py", line 18, in <module>
    from .user_service import UserService
  File "/ragflow/api/db/services/user_service.py", line 22, in <module>
    from api.db.db_models import DB, UserTenant
  File "/ragflow/api/db/db_models.py", line 32, in <module>
    from api.apps import app
  File "/ragflow/api/apps/__init__.py", line 26, in <module>
    from api.settings import DATABASE, stat_logger, SECRET_KEY
  File "/ragflow/api/settings.py", line 35, in <module>
    from rag.utils.es_conn import ELASTICSEARCH
  File "/ragflow/rag/utils/es_conn.py", line 460, in <module>
    ELASTICSEARCH = ESConnection()
    from api.db.db_models import close_connection
  File "/ragflow/rag/utils/__init__.py", line 12, in _singleton
  File "/ragflow/api/db/db_models.py", line 32, in <module>
    instances[key] = cls(*args, **kw)
  File "/ragflow/rag/utils/es_conn.py", line 24, in __init__
    raise Exception("Can't connect to ES cluster")

Is there any wrong from my operation? Thanks!

on the port :1200, looks Elastic search was up
{
name: "es01",
cluster_name: "docker-cluster",
cluster_uuid: "AApBy9PCQSKf_t_9eAP75A",
version: {
number: "8.11.3",
build_flavor: "default",
build_type: "docker",
build_hash: "64cf052f3b56b1fd4449f5454cb88aca7e739d9a",
build_date: "2023-12-08T11:33:53.634979452Z",
build_snapshot: false,
lucene_version: "9.8.0",
minimum_wire_compatibility_version: "7.17.0",
minimum_index_compatibility_version: "7.0.0"
},
tagline: "You Know, for Search"
}

KIBANA_PORT not up, all other ports were up

Can access webpage, but register failed.

**Comment by KevinHuSh** [comment]
If you're using  RAGFLOW_VERSION=v0.7.0, the code should also be 0.7.0.
My suggestion is to use dev version of docker images and pull the latest code.

**Comment by wuchewuche** [comment]
Mac-mini ragflow % docker compose -f docker/docker-compose.yml up -d
WARN[0000] The "HF_ENDPOINT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "MACOS" variable is not set. Defaulting to a blank string. 
[+] Running 12/14
 ⠹ ragflow Pulling                                                                                                      21.3s 
 ✔ mysql Pulled                                                                                                          4.2s 
 ⠹ minio [⣿⣿⣿⣄⠀⠀⠀] Pulling                                                                                              21.3s 
 ⠹ es01 [⣿⣿⣿⣿⠀⣿⣿⣿⣿⣿] 78.21MB / 490.3MB Pulling                                                                          21.3s 
                                                                                                                              
                                                                                                                              
                                                                                                                              
                                                                                                                              
                                                                                                                              
                                                                                                                              
                                                                                                                              
                                                                                                                              
                                                                                                                              
                                                                                                                              
no matching manifest for linux/arm64/v8 in the manifest list entries


what can i do? @KevinHuSh

**Reply by KevinHuSh** [reply]
Refer to [this](https://ragflow.io/docs/dev/build_docker_image).

**Comment by hall-ofame** [comment]
执行到docker compose -f docker/docker-compose.yml up -d显示Error response from daemon: {"message":"No such image: mysql:8.0.39"}，但是我本地已经配置成了8.0.39
 docker pull mysql:8.0.39
8.0.39: Pulling from library/mysql
Digest: sha256:ccb8f749bb5e59f9f8f03bf7282c7ef27a93a1814a24f0a8a926fb4e19b7fb97
Status: Image is up to date for mysql:8.0.39
docker.io/library/mysql:8.0.39
也是成功的，为什么还是不行呀

**Comment by wuchewuche** [comment]
I run git clone https://github.com/infiniflow/ragflow.git
cd ragflow/
pip3 install huggingface_hub nltk
python3 download_deps.py
docker build -f Dockerfile.deps -t infiniflow/ragflow_deps .
docker build -f Dockerfile -t infiniflow/ragflow:nightly .
docker build --build-arg LIGHTEN=1 -f Dockerfile -t infiniflow/ragflow:nightly-slim .
and    launchctl setenv OLLAMA_HOST "0.0.0.0:11434" and
Restart Ollama application
but docker logs -f ragflow-server run unsuccessful @KevinHuSh how can i do next？

**Comment by KevinHuSh** [comment]
Please refer to [this](https://ragflow.io/docs/dev/).

**Comment by wuchewuche** [comment]
hey @KevinHuSh  i have some warning in this.
wuche@wuchedeMac-mini ragflow % docker compose -f docker/docker-compose.yml up -d
WARN[0000] The "HF_ENDPOINT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "MACOS" variable is not set. Defaulting to a blank string. 
[+] Running 20/21
 ✔ mysql Pulled                                   5.3s 
 ✔ minio Pulled                                  24.5s 
   ✔ b7b0bb81ef0e Pull complete                  10.7s 
   ✔ c1e42aa8a275 Pull complete                  10.7s 
   ✔ e9b6753743c5 Pull complete                  16.1s 
   ✔ 014647d6ee36 Pull complete                  18.2s 
   ✔ 145535a1571c Pull complete                  18.2s 
   ✔ a401bbb8b028 Pull complete                  18.6s 
   ✔ abe03354e1ec Pull complete                  19.4s 
 ✔ es01 Pulled                                   39.5s 
   ✔ 572aee731353 Pull complete                   4.9s 
   ✔ 124cb7e3e37e Pull complete                   5.2s 
   ✔ 983416f418c0 Pull complete                   5.2s 
   ✔ 89732bc75041 Pull complete                   5.2s 
   ✔ 8c785eaec661 Pull complete                  34.9s 
   ✔ eea140bef66b Pull complete                  34.9s 
   ✔ d16bcad91132 Pull complete                  34.9s 
   ✔ c23208337cf8 Pull complete                  35.0s 
   ✔ 2d6112686745 Pull complete                  35.0s 
   ✔ e6fc0d805b6e Pull complete                  35.0s 
 ⠙ ragflow Pulling                               64.1s 
no matching manifest for linux/arm64/v8 in the manifest list entries
wuche@wuchedeMac-mini ragflow %   what can i do?

**Reply by KevinHuSh** [reply]
Ignore warning messages.

**Comment by zhangguoliang-666** [comment]
time="2025-02-13T22:57:39+08:00" level=warning msg="The \"HF_ENDPOINT\" variable is not set. Defaulting to a blank string."
time="2025-02-13T22:57:39+08:00" level=warning msg="The \"MACOS\" variable is not set. Defaulting to a blank string."
[+] Running 4/12
 ✘ redis Error         Get "https://registry-1.docker.io/v2/": net/http: request canceled while wa...             15.2s
 ✘ ragflow Error       context canceled                                                                           15.2s
 ✘ es01 Error          context canceled                                                                           15.2s
 - minio [⠀⣦⠀⠀⠀⠀⠀] Pulling                                                                                        15.2s
   - f72461870632 Downloading [>                                                  ]  75.43kB...                   10.2s
   - 683391db8929 Downloading [====================================>              ]  91.08kB...                   10.2s
   - ba8b8055313f Pulling fs layer                                                                                10.2s
   - a8e0787fb7ed Waiting                                                                                         10.2s
   - fd20cadb8d39 Waiting                                                                                         10.2s
   - 3738ac54d510 Waiting                                                                                         10.2s
   - 128c59a31db4 Waiting                                                                                         10.2s
 ✘ mysql Error         context canceled                                                                           15.2s
Error response from daemon: Get "https://registry-1.docker.io/v2/": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
that's why? help me  please
